src/neo/Network/P2P/TaskManager.cs

```diff
@@ -273,8 +273,10 @@ private void RequestTasks(IActorRef remoteNode, TaskSession session)
                 }
                 if (!sessions.Values.Any(p => p.IndexTasks.ContainsKey(startHeight) && p.IndexTasks[startHeight].AddSeconds(5) > TimeProvider.Current.UtcNow))
                 {
-                    session.IndexTasks[startHeight] = TimeProvider.Current.UtcNow;
-                    remoteNode.Tell(Message.Create(MessageCommand.GetBlockByIndex, GetBlockByIndexPayload.Create(startHeight)));
+                    for (uint i = startHeight; i < session.LastBlockIndex; i++)
+                        session.IndexTasks[i] = TimeProvider.Current.UtcNow;
+                    short count = (short)(session.LastBlockIndex - startHeight);
+                    remoteNode.Tell(Message.Create(MessageCommand.GetBlockByIndex, GetBlockByIndexPayload.Create(startHeight, count)));
```

Comment:
You need to check `HeadersPayload.MaxHeadersCount`.

Reply:
Done.